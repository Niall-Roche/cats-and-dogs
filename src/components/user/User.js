import {shape, string} from 'prop-types'
import {styled} from 'styled-components'
import Image from '../image/Image'
import {GIPHY_ICON, INSTAGRAM_ICON} from '../../constants/images'

const propTypes = {
  data: shape({
    avatar_url: string,
    description: string,
    instagram_url: string,
    profile_url: string,
    username: string,
  }).isRequired,
}

const UserContainer = styled.div`
  align-self: start;
  background-color: ${props => props?.theme?.secondary};
  border-radius: 18px;
`

const UserDetails = styled.div`
  background-color: ${props => props?.theme?.accent};
  border-radius: 18px;
  padding: 10px;
`

const ProfileImage = styled(Image)`
  object-fit: cover;
  width: 50px;
  height: 50px;
  margin-right: 5px;
`

const UserLinks = styled.div`
  margin-top: 5px;
  display: flex;
  gap: 10px;
`

const UserLink = styled.a`
  text-decoration: none;
  color: ${props => props?.theme?.primary};
  transition: opacity .2s;
  &:hover {
    opacity: 0.7;
  }
`

const UserName = styled.span`
  font-weight: bold;
  font-size: 14px;
`

const ProfileDescription = styled.p`
  margin-left: 10px;
  margin-right: 10px;
  max-width: 250px;
  white-space: pre-wrap;
  font-size: 14px;
`

const User = ({data}) => {
  return (
    <UserContainer>
      <UserDetails className='d-flex'>
        <ProfileImage src={data?.avatar_url} />
        <div>
          <UserName>{data?.username}</UserName>
          <UserLinks>
            <UserLink target='blank' rel='noreferrer' href={data?.profile_url}>
              <img src={GIPHY_ICON} height={25} width={25} />
            </UserLink>
            <UserLink target='blank' rel='noreferrer' href={data?.instagram_url}>
              <img src={INSTAGRAM_ICON} height={25} width={25} />
            </UserLink>
          </UserLinks>
        </div>
      </UserDetails>
      <ProfileDescription>{data?.description}</ProfileDescription>
    </UserContainer>
  )
}

User.propTypes = propTypes

export default User