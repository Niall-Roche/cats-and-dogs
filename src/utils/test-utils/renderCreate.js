import {ThemeProvider} from '@tanstack/react-query-devtools/build/lib/theme'
import TestRenderer from 'react-test-renderer'
import theme from '../../app-theme.json'
import {CatsAndDogsProvider} from '../../hooks/useCatsAndDogs'
import {PaginationProvider} from '../../hooks/usePagination'

export default comp => TestRenderer.create(
  <CatsAndDogsProvider>
    <PaginationProvider>
      <ThemeProvider theme={theme}>
        {comp}
      </ThemeProvider>
    </PaginationProvider>
  </CatsAndDogsProvider>
)