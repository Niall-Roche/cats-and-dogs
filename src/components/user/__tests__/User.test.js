import React from 'react'
import User from '../User'
import 'jest-styled-components'
import renderCreate from '../../../utils/test-utils/renderCreate'

describe('User', () => {
  it('should match snapshot', () => {
    const comp = renderCreate(
      <User
        data={{
          avatar_url: 'https://media3.giphy.com/avatars/bestfriends/4Ja63d50vmML.gif',
          description: 'DESCRIPTION',
          instagram_url: 'https://www.instagram.com/bestfriendsanimalsociety/',
          profile_url: 'https://giphy.com/bestfriends/',
          username: 'bestfriends',
        }}
      />
    )
    const tree = comp.toJSON()

    expect(tree).toMatchSnapshot()
  })
})