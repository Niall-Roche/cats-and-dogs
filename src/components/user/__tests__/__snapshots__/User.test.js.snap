// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should match snapshot 1`] = `
<div
  className="hUMCjG"
>
  <div
    className="kbLjDT d-flex"
  >
    <div
      className="kBnjTn"
    >
      <img
        className="lbBMSi cKDFqv"
        onLoad={[Function]}
        src="https://media3.giphy.com/avatars/bestfriends/4Ja63d50vmML.gif"
      />
    </div>
    <div>
      <span
        className="bWaHCS"
      >
        bestfriends
      </span>
      <div
        className="ecapwf"
      >
        <a
          className="jbTKNC"
          href="https://giphy.com/bestfriends/"
          rel="noreferrer"
          target="blank"
        >
          <img
            height={25}
            src="giphy-icon.png"
            width={25}
          />
        </a>
        <a
          className="jbTKNC"
          href="https://www.instagram.com/bestfriendsanimalsociety/"
          rel="noreferrer"
          target="blank"
        >
          <img
            height={25}
            src="instagram-icon.png"
            width={25}
          />
        </a>
      </div>
    </div>
  </div>
  <p
    className="cdWVoH"
  >
    DESCRIPTION
  </p>
</div>
`;
